core/arch/riscv/include/encoding.h

```diff
@@ -0,0 +1,4830 @@
+/* SPDX-License-Identifier: BSD-4-Clause-UC */
```

Comment:
Does this file have an upstream? Is it copied from somewhere?

How about calling this file `riscv.h` instead? It seems to serve more or less the same purpose as the `arm.h` file.